pkg/controller/resource_sync_task.go

```diff
@@ -82,6 +84,9 @@ func (st *resourceSyncTask) processResource(res *smith_v1.Resource) resourceInfo
 
 	// Do as much prevalidation of the spec as we can before dependencies are resolved.
 	// (e.g. plugin/service instance/service binding schemas)
+	// We may want to move this out of the resource processing entirely and do
+	// it before we start processing the bundle, both to fail faster and to avoid
+	// many schema validations.
```

Comment:
Ah, well, I guess for a naive implementation it mostly helps in the failure case (bundle reprocessing won't happen as many times since the bundle resources won't be changing), but to do it 'properly' I was imagining having a 'ResourceVersion' equivalent (cf the plan caching - would we need an admission controller to set this?) and mark when we've done the validation for that particular bundle. Though ultimately that's not really different from just caching the results of the validation for a particular bundle resource, it's just that we don't have the ugliness of having to compare the parameters block.

But my preference, ultimately, would be to dump the plan cache and instead have only a cache that looking like:

namespace/bundlename/resourcename -> resourceblock/validationresult

Then do a lookup of the resource and a comparison of the resourceblock each time to determine if we can reuse the validation result.

Anyway, as you said, we can solve the problem if it's actually a problem. I'll remove this comment.